Feature/#19 대회 진행 상태 토글 및 현재 진행 중인 대회 조회 api 구현

myeowon opened the pull request:
## 🔥 연관된 이슈

close: #19 

## 📜 작업 내용

- 대회 진행 상태 토글 api를 구현했습니다.
- 현재 진행 중인 대회 조회 api를 구현했습니다.
- Contest 엔티티에 bannerId를 추가했습니다. 성재님이 대회 배너 구현해주실 때 수정해야할 부분 있다면 수정해주시면 될 것 같습니다!

## 💬 리뷰 요구사항

- 리뷰 요구사항을 작성해주세요.

## ✨ 기타

- 오늘의 한마디

myeowon committed:
fix : dto

myeowon committed:
fix : 대회 진행 상태 토글

myeowon committed:
fix : 현재 진행 중인 대회 조회

sjmoon00 commented on src/main/java/com/opus/opus/modules/contest/domain/Contest.java:
```diff
@@ -43,15 +46,22 @@ public class Contest extends BaseEntity {
 
+    @Column
+    private Long bannerId;
+
```
파일에서 대회id를 가지고 있어서 여기엔 없어도 괜찮을것 같습니다

myeowon replied:
넵 알겠습니다!

sjmoon00 reviewed (approved):
수고하셨습니다~

JJimini commented on src/main/java/com/opus/opus/modules/contest/application/ContestCommandService.java:
```diff
@@ -8,4 +16,39 @@
+        if (currentValue.equals(requestValue)) {
+            if (currentValue) {
+                throw new ContestException(ALREADY_CURRENT_CONTEST);
+            } else {
+                throw new ContestException(ALREADY_NOT_CURRENT_CONTEST);
+            }
+        }
+    }
```
```suggestion
private void validateSameCurrentRequest(final Boolean currentValue, final Boolean requestValue) {
    if (currentValue == requestValue) {
        throw new ContestException(
                currentValue ? ALREADY_CURRENT_CONTEST : ALREADY_NOT_CURRENT_CONTEST
        );
    }
}
```

JJimini commented on src/main/java/com/opus/opus/modules/contest/application/ContestCommandService.java:
```diff
@@ -8,4 +16,39 @@
+        // 변경
+        contest.updateIsCurrent(isCurrent);
+        String message = isCurrent ? "현재 대회로 등록되었습니다." : "현재 대회 지정이 해제되었습니다.";
+        return ContestCurrentToggleResponse.of(contest.getId(), isCurrent, message);
```
message를 service에서 처리해도 상관은 없을 것 같은데, of 메서드에서 처리하는 방법도 있을 것 같아용

JJimini commented on src/main/java/com/opus/opus/modules/contest/application/dto/request/ContestCurrentToggleRequest.java:
```diff
@@ -0,0 +1,8 @@
+package com.opus.opus.modules.contest.application.dto.request;
+
+import jakarta.validation.constraints.NotNull;
+
+public record ContestCurrentToggleRequest(
+        @NotNull Boolean isCurrent
+) {
```
nit: 필드를 NotNull 밑에 적어주시면 좋을 것 같아요!

JJimini reviewed (approved):
수고하셨습니다~

myeowon committed:
chore : 충돌 해결

myeowon committed:
feat : 대회 현재 상태 관련 컨트롤러

# Conflicts:
#	src/main/java/com/opus/opus/modules/contest/api/ContestController.java

myeowon committed:
chore : 충돌 해결

myeowon committed:
fix : 리뷰 반영

myeowon committed:
fix : 대회 배너 제거
